src/shared/mount-util.c

```diff
@@ -1049,14 +1049,42 @@ int make_mount_point(const char *path) {
         return 1;
 }
 
-static int make_userns(uid_t uid_shift, uid_t uid_range) {
-        char line[DECIMAL_STR_MAX(uid_t)*3+3+1];
```

Comment:
The size is still fixed, just at twice the length - given asprintf is expensive (or so I'm often told) and this is a library call, maybe just doubling the static buffer and optionally pointing a second xsprintf past the first newline would work? But it's not too important I guess

Reply:
Well, depends on the flag. But this is not called in an inner loop, but exactly once. I think performance doesn't really matter here, but it is a lot more readable and less likely to mistakes if we juse use dynamic memory.